Assets/Scripts/Infrastructure/UpdateRunner.cs

```diff
@@ -56,7 +57,7 @@ public void Subscribe(Action<float> onUpdate, float updatePeriod)
                 {
                     if (m_Subscribers.Add(onUpdate))
                     {
-                        m_SubscriberData.Add(onUpdate, new SubscriberData() { Period = updatePeriod, NextCallTime = 0 });
+                        m_SubscriberData.Add(onUpdate, new SubscriberData() { Period = updatePeriod, NextCallTime = 0, LastCallTime = Time.time });

```

Comment:
is this fixing the useless lobby updates issue you were talking about in your jira comments?

Reply:
If so I'd try to mention this our changelog, just so users know to update their code for those that took inspiration from our code. They'll need to update their own code to make sure they don't do the same mistake

Reply:
No, this addresses the issue causing the Heartbeat messages not being sent at the desired frequency to the Lobby service (mentioned in the PR description).
The unneeded updates issue was caused by us receiving the Lobby updates and systematically marking those as changing our local lobbies, so we would also send updates to Lobby each time we received an update, even if no change occurred. This is a nice catch though, because it seems I did not create a bug for this issue. This PR doesn't fix the issue, just greatly reduces its impact. I'll create a Jira ticket for it

Reply:
[here](https://jira.unity3d.com/browse/MTT-5665) it is